packages/client/src/util/streamutils.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const START_TOKEN = new Uint8Array([123, 34, 100, 97, 116, 97, 34, 58, 91]); // `{"data":[`
+const OBJECT_OPEN_CHAR_CODE = 123; // '{'
+const OBJECT_CLOSE_CHAR_CODE = 125; // '}'
+
+export async function* parseStreamedResponse(
```

Comment:
Pulled this parsing helper from the legacy implementation

Reply:
Cool with this for now as we know it works but I'm curious if there is something a bit less custom that works out of the box for browsers? Def for a follow up.